cypress/integration/articleFlows/postSidebarActions.spec.js

```diff
@@ -0,0 +1,33 @@
+describe('Post sidebar actions', () => {
+  beforeEach(() => {
+    cy.testSetup();
+    cy.fixture('users/articleEditorV2User.json').as('user');
+
+    cy.get('@user').then((user) => {
+      cy.loginUser(user).then(() => {
+        cy.createArticle({
+          title: 'Test Article',
+          tags: ['beginner', 'ruby', 'go'],
+          content: `This is a test article's contents.`,
+          published: true,
+        }).then((response) => {
+          cy.visit(response.body.current_state_path);
+        });
+      });
+    });
+  });
+
+  it('should open the share menu for the post', () => {
+    // TODO: Get this working
+    cy.findByRole('button', { name: /^Toggle dropdown menu$/i }).click();
```

Comment:
The click works but does not open the menu. It works in the app and when you click on it after the test fails. I even tried checking if the event was attached by doing

```
cy.get(
	'button[aria-label="Toggle dropdown menu"][data-initialized="true"]',
).click();
```

instead. Not sure what's up.

Reply:
I had a dig into this, and what's happening is that in the button click handler we:

- add the class "block" to the dropdown, and 
- we attach the clickOutsideListener to the the document....

 **but**  the initial button click event triggers the clickOutsideListener also when we're running in Cypress 🙃  Basically there is something funky going on with the event propagation, and the dropdown is opened and immediately closed all on that toggle button click.

The clickOutsideListener checks for various elements to see whether it should close dropdowns or not, but the `target` it receives on that first button click is the `<path>` element inside the icon svg, which wasn't accounted for in that method.

To be honest, I'm still trying to get my head around it. It doesn't happen during normal usage, and I think it is _something_ to do with how Cypress is actioning that click. 

To fix it we can add another check to the `shouldCloseDropdown` function to make sure the path within the `dropdown-icon` is also ignored:

```js
return !(
      event.target.matches('.dropdown-icon') ||
      event.target.parentElement.classList.contains('dropdown-icon') ||
      event.target.matches('.dropbtn') ||
      event.target.matches('clipboard-copy') ||
      isCopyIconChild ||
      event.target.parentElement.classList.contains('dropdown-link-row')
    );
```

I also noticed though that changing the line in the cypress test to

```js
cy.findByRole('button', { name: /^Toggle dropdown menu$/i }).click({ force: true});
```

also fixes the issue. Further evidence that there's something fundamental about how cypress executes click events that's causing this weirdness.

Reply:
Thanks for digging. Just up before working out, so I've made your suggested changes and popped this to ready for review.